sim/src/recorder.rs

```diff
@@ -29,52 +29,87 @@ impl TrafficRecorder {
 
     pub fn handle_event(&mut self, time: Time, ev: &Event, map: &Map, driving: &DrivingSimState) {
         if let Event::AgentEntersTraversable(AgentID::Car(car), Some(trip), on, _) = ev {
-            if self.seen_trips.contains(trip) {
-                return;
-            }
-            if let Traversable::Lane(l) = on {
-                if self.capture_points.contains(&map.get_l(*l).src_i) {
-                    // Where do they exit?
-                    for step in driving.get_path(*car).unwrap().get_steps() {
-                        if let PathStep::Turn(t) = step {
-                            if self.capture_points.contains(&t.parent) {
-                                self.trips.push(IndividTrip::new(
-                                    time,
-                                    TripPurpose::Shopping,
-                                    TripEndpoint::SuddenlyAppear(Position::start(*l)),
-                                    TripEndpoint::Border(t.parent),
-                                    if car.vehicle_type == VehicleType::Bike {
-                                        TripMode::Bike
-                                    } else {
-                                        TripMode::Drive
-                                    },
-                                ));
-                                self.seen_trips.insert(*trip);
-                                return;
-                            }
-                        }
+            self.on_car_enters_traversable(time, &car, &trip, &on, map, driving);
+        }
+    }
+
+    fn on_car_enters_traversable(
+        &mut self,
+        time: Time,
+        car: &CarID,
+        trip: &TripID,
+        on: &Traversable,
+        map: &Map,
+        driving: &DrivingSimState,
+    ) {
+        if self.seen_trips.contains(trip) {
+            return;
+        }
+        if let Traversable::Lane(lane) = on {
+            self.on_car_enters_lane(time, &car, &trip, &lane, map, driving);
+        }
+    }
+
+    fn on_car_enters_lane(
+        &mut self,
+        time: Time,
+        car: &CarID,
+        trip: &TripID,
+        lane: &LaneID,
+        map: &Map,
+        driving: &DrivingSimState,
+    ) {
+        if !self.capture_points.contains(&map.get_l(*lane).src_i) {
+            return;
+        }
+        // Where do they exit?
+        let intersections = driving
+            .get_path(*car)
+            .unwrap()
+            .get_steps()
+            .iter()
+            .filter_map(|step| {
```

Comment:
The previous code stopped at the first match... this is looking for the one intersection where the vehicle exits the area being recorded. I don't think we want to `filter_map` and potentially wind up with multiple